pkg/kinflate/resource/util.go

```diff
@@ -90,3 +90,61 @@ func Merge(rcs ...ResourceCollection) (ResourceCollection, error) {
 
 	return all, nil
 }
+
+// Merge will merge all of the entries in the slice of ResourceCollection with Override
+// If there is already an entry with the same GVKN exists, different actions are performed according to value of Behavior field
+// 'create': create a new one;
+// 'replace': replace the data only; keep the labels and annotations
+// 'merge': merge the data; keep the labels and annotations
+func MergeWithOverride(rcs ...ResourceCollection) (ResourceCollection, error) {
+	all := ResourceCollection{}
+
+	for _, rc := range rcs {
+		for gvkn, obj := range rc {
+			if _, found := all[gvkn]; found {
+				switch all[gvkn].Behavior {
+				case "", "create":
+					return nil, fmt.Errorf("Create an existing gvkn %#v is not allowed", gvkn)
+				case "replace":
+					all[gvkn].Data.SetLabels(mergeMap(obj.Data.GetLabels(), all[gvkn].Data.GetLabels()))
+					all[gvkn].Data.SetAnnotations(mergeMap(obj.Data.GetAnnotations(), all[gvkn].Data.GetAnnotations()))
+					all[gvkn].Data.SetName(obj.Data.GetName())
+				case "merge":
+					all[gvkn].Data.SetLabels(mergeMap(obj.Data.GetLabels(), all[gvkn].Data.GetLabels()))
+					all[gvkn].Data.SetAnnotations(mergeMap(obj.Data.GetAnnotations(), all[gvkn].Data.GetAnnotations()))
+					all[gvkn].Data.SetName(obj.Data.GetName())
+					mergeData(all[gvkn].Data.Object, obj.Data.Object)
+				default:
+					return nil, fmt.Errorf("The behavior of %#v must be one of create, merge and replace", gvkn)
+				}
+			} else {
+				all[gvkn] = obj
+				obj.Data.GetAnnotations()
+			}
+		}
+	}
+	return all, nil
+}
+
+func mergeMap(maps ...map[string]string) map[string]string {
+	mergedMap := maps[0]
+	for _, m := range maps[1:] {
+		for key, value := range m {
+			mergedMap[key] = value
+		}
+	}
+	return mergedMap
+}
+
+func mergeData(mergedTo map[string]interface{}, mergedFrom map[string]interface{}) {
+	mergedToMap, toOK := mergedTo["data"].(map[string]interface{})
```

Comment:
We should try to avoid using a hardcoded string key here. It should  be possible. :)

Reply:
yes, we should avoid this. Let's discuss offline.